### PR TITLE
Support @material-ui/core v1.0.0-rc.1

### PR DESCRIPTION
**Pull request by mihirsoni**
Fix errors shown before touch/submit and add tests for mapError function

Update material-ui to v1.0.0 final

Fix broken package.json

Fix styling issues

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,15 +51,15 @@ and then pass the component class directly to the `component` prop of `Field`.
 
 ```js
 import { reduxForm, Field } from 'redux-form'
-import MenuItem from 'material-ui/Menu'
-import { Radio } from 'material-ui/Radio'
-import { FormControlLabel } from 'material-ui/Form'
+import MenuItem from '@material-ui/core/MenuItem'
+import RadioButton from '@material-ui/core/RadioButton'
```
Comment (andreapier) [inline README.md]: This should be `import Radio from '@material-ui/core/RadioButton'` 
Also check `RadioButton` usages a few lines below.

File: README.md [hunk 2]
```diff
@@ -51,15 +51,15 @@ and then pass the component class directly to the `component` prop of `Field`.
...
 import {
   Checkbox,
   RadioGroup,
   Select,
   TextField,
   Switch,
+  FormControlLabel,
```
Comment (andreapier) [inline README.md:62]: This should not be imported from redux-form-material-ui but from material-ui

File: package.json [hunk 3]
```diff
@@ -61,43 +62,32 @@
...
     "mocha": "^3.5.0",
     "mocha-lcov-reporter": "^1.3.0",
     "nyc": "^11.1.0",
     "prettier": "^1.5.3",
-    "react": "^15.6.1",
-    "react-addons-test-utils": "^15.6.0",
-    "react-dom": "^15.6.1",
+    "react": "^16.3.2",
+    "react-dom": "^16.3.2",
     "react-tap-event-plugin": "^2.0.1",
```
Comment (andreapier) [inline package.json]: I think it's better to remove this, as it is giving warning while installing dependencies:  
```
warning " > react-tap-event-plugin@2.0.1" has incorrect peer dependency "react@^15.4.0-0".
warning " > react-tap-event-plugin@2.0.1" has incorrect peer dependency "react-dom@^15.4.0-0".
```
I asked for advice in material-ui's gitter room and it seems this package is not required any more. See [here](https://gitter.im/mui-org/material-ui?at=5b03ed389ad18967b905d534).

**Review by andreapier — COMMENTED** [review]
This looks good to me.  There is no blocking issues, but I left 3 comments. I think support for v1.0.0 is better that way.  
I made those changes while resolving conflicts so I guess they went unnoticed.  
Howerver, if you don't want to make them you can merge this branch as it is.... It is still a major improvement for people using v1.0.0.

File: package.json (hunk 3 above)
Comment (mihirsoni) [reply package.json]: I should be removing an and running tests :)

**Comment by andreapier** [comment]
@mihirsoni Are you going to merge this PR?

File: README.md (hunk 2 above)
Comment (andidev) [reply README.md:62]: This is still not fixed in docs

**Pull request merged**